### ISSUE TITLE
Hide the history of the events of a ticket

### ISSUE THREAD

**Issue by marien-probesys** [Probesys/bileto#13]
As **technical**,  
I want to **hide events in the history of a ticket**,  
so **I can see more clearly the answers.**

**Comment by marien-probesys** [comment]
There are few questions to answer for this ticket.

The first one is: do we want to remember that the user hid the events, i.e. do we need to store the preference in DB/local storage or simply pass the information in the URL?

→ I feel like this option will be considered/used as a preference for the users. As a user, I expect this info to be remembered if I hide the events in a ticket. The main issue is that users can forget about the events. It can be put into perspective with the fact that events are not always useful and that users will remember them when they'll need them.

In this context, we can ask two other questions:

1. should we hide events globally (i.e. hiding the events in a ticket will hide events in other tickets), or per ticket? (i.e. hiding the events in a ticket will not impact other tickets)
2. should we hide events per device? (e.g. hiding events on PC will not hide them on mobile)

If we choose to hide the events per ticket, it relativizes the fact that users can forget about events, but it can be a bit more complicated to implement. I think it's fine to hide events globally. It particularly makes sense if the users consider this option as a preference. This is what Gitlab does btw (while I'm not fan of their UX).

Hiding events globally per device seems to be the simplest approach: it only requires to store a boolean in the local storage of the browser and I don't think it's too disturbing to the users since 1. they'll probably mostly use Bileto on a single device and 2. they'll probably don't even notice that there is a difference in the preferences between their devices.

On the other hand, it adds a new way to store users' preferences (local storage vs. database) and it can disturb developers. Also, adding a new property in User entity is not too complicated and will not take that much time. Also, having the option server-side will allow to improve the performance since the server will not have to load the events at all.

In conclusion, my preference goes to saving the option in database and globally for all the tickets.
